crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -248,6 +276,56 @@ impl WellKnownNote {
                     }
                 }
             },
+            WellKnownNote::P2IDE => {
+                let note_inputs = note.inputs().values();
+                // check the expected number of inputs
+                if note_inputs.len() != self.num_expected_inputs() {
+                    return NoteAccountCompatibility::No;
+                }
+
+                let Some(input_account_id) = try_read_account_id_from_inputs(note_inputs) else {
+                    return NoteAccountCompatibility::No;
+                };
+                let sender_account_id = note.metadata().sender();
+                let is_target = input_account_id == target_account_id;
+                let is_sender = input_account_id == sender_account_id;
+
+                // only target or sender may consume
+                if !is_target && !is_sender {
+                    return NoteAccountCompatibility::No;
+                }
+
+                // parse recall height
+                let recall_height: Result<u32, _> = note_inputs[2].try_into();
+                let Ok(recall_height) = recall_height else {
+                    return NoteAccountCompatibility::No;
+                };
+                // parse timelock height
+                let timelock_height: Result<u32, _> = note_inputs[3].try_into();
+                let Ok(timelock_height) = timelock_height else {
+                    return NoteAccountCompatibility::No;
+                };
+
+                // timelock check
+                if block_ref.as_u32() <= timelock_height {
```

Comment:
The condition for successful consumption is `timelock_block_height <= current_block_height` so to check if the consumption will fail, don't we have to check for `timelock_block_height > current_block_height` here instead of `timelock_block_height >= current_block_height`?